pkg/generate/app/cmd/newapp.go

```diff
@@ -586,7 +586,18 @@ func (c *AppConfig) validate() (app.Environment, app.Environment, error) {
 		return nil
 	})
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
+	}
+	buildEnv, err := app.ParseAndCombineEnvironment(c.BuildEnvironment, c.BuildEnvironmentFiles, c.In, func(key, file string) error {
```

Comment:
anything passed to --env is being stored in the &config.BuildEnvironment, which is used as c.BuildEnvironment here.